batchflow/models/tf/vnet.py

```diff
@@ -1,158 +1,111 @@
+        f_0 = initial_block_filters[-1] if isinstance(initial_block_filters, (list, tuple)) else initial_block_filters
 
-    @classmethod
-    def decoder_block(cls, inputs, name, **kwargs):
-        """ 2x2x2 transposed convolution + 5x5x5 convolutions
-
-        Parameters
-        ----------
-        inputs : tf.Tensor
-            input tensor
-        name : str
-            scope name
-
-        Returns
-        -------
-        tf.Tensor
-        """
-        kwargs = cls.fill_params('body', **kwargs)
-        layout, filters, kernel_size = cls.pop(['layout', 'filters', 'kernel_size'], kwargs)
-        upsample_args = cls.pop('upsample', kwargs)
-
-        with tf.variable_scope(name):
-            x, skip = inputs
-            inputs = None
-            x = cls.upsample(x, filters=filters, name='upsample', **upsample_args, **kwargs)
-            x = cls.crop(x, skip, data_format=kwargs.get('data_format'))
-            axis = cls.channels_axis(kwargs.get('data_format'))
-            x = tf.concat((skip, x), axis=axis)
-            x = ResNet.block(x, layout=layout, filters=filters, kernel_size=kernel_size, downsample=0,
-                             name='conv', **kwargs)
-        return x
+        encoder_filters = [f_0*2**(i+1) for i in range(num_stages)]
+        decoder_filters = encoder_filters[::-1]
 
-    @classmethod
-    def head(cls, inputs, num_classes, name='head', **kwargs):
-        """ 1x1 convolution
-
-        Parameters
-        ----------
-        inputs : tf.Tensor
-            input tensor
-        num_classes : int
-            number of classes (and number of filters in the last 1x1 convolution)
-        name : str
-            scope name
-
-        Returns
-        -------
-        tf.Tensor
-        """
-        kwargs = cls.fill_params('head', **kwargs)
-        x = conv_block(inputs, filters=num_classes, name=name, **kwargs)
-        return x
+        if num_stages != 4:
+            if config.get('body/encoder/blocks/layout') == ['cna'*2] + ['cna'*3] * 3:
+                config['body/encoder/blocks/layout'] = ['cna'*2 if i < 1 else 'cna'*3 for i in range(num_stages)]
+
+            if config.get('body/encoder/blocks/filters') == [32, 64, 128, 256]:
+                config['body/encoder/blocks/filters'] = encoder_filters
+
+            if config.get('body/encoder/downsample/filters') == [32, 64, 128, 256]:
+                config['body/encoder/downsample/filters'] = encoder_filters
+
+            if config.get('body/decoder/blocks/layout') == ['cna'*3] * 3 + ['cna'*2]:
+                config['body/decoder/blocks/layout'] = ['cna'*3 for i in range(num_stages-1)] + ['cna'*2]
+
+            if config.get('body/decoder/blocks/filters') == [256, 128, 64, 32]:
+                config['body/decoder/blocks/filters'] = decoder_filters
+
+            if config.get('head/filters') == [48, 32]:
+                config['head/filters'] = [f_0 * 3, f_0 * 2]
+
+        return config
```

Comment:
Perhaps it might, but let's look at the code to decide.
What I like about the current implementation is that all the model configuration is gathered in one place thus making it easy to grasp and fix when necessary.